### PR TITLE
Added support for RoCE - zVM HCP 

### PR DESCRIPTION
**Pull request by veera-damisetti**
- Added support for RoCE - zVM HCP 
- Updated Deletion procedure for HCP 
- Updated parm files  for HCP zVM

### CODE REVIEW

**Review by AmadeusPodvratnik — COMMENTED** [review]
Hi @veera-damisetti , thank you for the PR. I'm just thinking if it's worth to mention the supported network options in the documentation. At the moment only Hypersocket is missing.
Could you pls add the supported network interfaces to your hcp docu. Thx.

**Comment by veera-damisetti** [comment]
@AmadeusPodvratnik Updated the documentation 
Thanks for the suggestion

**Comment by veera-damisetti** [comment]
@AmadeusPodvratnik Updated PR with new changes and verified with pipeline also 
Can you please review 
thanks

File: roles/scale_nodepool_and_wait_for_workers_hypershift/tasks/main.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@
   when: hypershift.compute_node_type | lower != 'zvm'
 
 - name: Patch Agents
-  shell: oc -n {{ hypershift.hcp.clusters_namespace }}-{{ hypershift.hcp.hosted_cluster_name }}  patch agent {{ agents[item] }} -p '{"spec":{"installation_disk_id":"/dev/vda","approved":true,"hostname":"worker-{{item}}.{{ hypershift.hcp.hosted_cluster_name }}.{{ hypershift.hcp.basedomain }}"}}' --type merge
+  shell: oc -n {{ hypershift.hcp.clusters_namespace }}-{{ hypershift.hcp.hosted_cluster_name }}  patch agent {{ agents[item] }} -p '{"spec":{"installation_disk_id":"/dev/vda","approved":true,"hostname":"compute-{{item}}.{{ hypershift.hcp.hosted_cluster_name }}.{{ hypershift.hcp.basedomain }}"}}' --type merge
```
Comment (AmadeusPodvratnik) [inline roles/scale_nodepool_and_wait_for_workers_hypershift/tasks/main.yaml:30]: Hi @veera-damisetti , you changed the name of the hostname to compute. Do you want to change the role to scale_nodepool_and_wait_for_compute_hypershift ? Just to make it consistent? If not ... np.

**Review by AmadeusPodvratnik — COMMENTED** [review]
Hi @veera-damisetti thx for the PR. Pls find my question.

File: roles/scale_nodepool_and_wait_for_workers_hypershift/tasks/main.yaml (hunk 1 above)
Comment (veera-damisetti) [reply roles/scale_nodepool_and_wait_for_workers_hypershift/tasks/main.yaml:30]: yeah sounds good...!
Updated.. thanks

**Review by AmadeusPodvratnik — APPROVED** [review]
Looks good

**Comment by veera-damisetti** [comment]
Thanks @AmadeusPodvratnik  for the quick review

**Pull request merged**